openblas: fix build on riscv64

ktpedre opened the pull request:
OpenBLAS now has support for the riscv64 architecture. This commit
extends the spack openblas package.py to handle building on riscv64.

ktpedre committed:
openblas: fix build on riscv64
OpenBLAS now has support for the riscv64 architecture. This commit
extends the spack openblas package.py to handle building on riscv64.

tldahlgren commented on var/spack/repos/builtin/packages/openblas/package.py:
```diff
@@ -220,7 +220,7 @@ def _microarch_target_args(self):
                 args.append('ARCH=' + arch_name)
 
-        if microarch.vendor == 'generic':
+        if microarch.vendor == 'generic' and microarch.family.name != 'riscv64':
```
Why use `micro arch.family.name` here and `micro arch.name` at line 242?

ktpedre replied:
@tldahlgren -- the thought was that there might someday be multiple archspec riscv64 uarchs that map to the "vendor=generic" case (e.g., a generic riscv64+rvv case). Currently, none of the riscv64 "generic" uarchs should map to the "DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 TARGET=GENERIC" case. 

Then, the "elif microarch.name == 'riscv64'" below was only meant to rename the archspec "riscv64" uarch to what OpenBLAS uses for this case, "RISCV64_GENERIC". All of the other cases will fall through to the else, where (ideally) the archspec uarch name matches the openblas target name.

All that said, currently it would work equally well to use the same "microarch.name" in both places, as currently there is only one vendor=generic riscv64 uarch in archspec-json. Happy to make that change, just let me know.

tldahlgren replied:
Sounds reasonable.  Please make them the same so the code is consistent at this time and the purpose for using of `microarch.family.name` will be clearer when there are more cases reflected in the code.  

Alternatively, you could add a comment to help explain the difference.

ktpedre replied:
@tldahlgren I made this change and verified that "spack install openblas" still works the same as before. Also, "spack install hpl" works using the built openblas and HPL runs successfully :)

ktpedre committed:
Use microarch.name rather than microarch.family.name.
This change was made to match the "elif microarch.name == 'riscv64'"
usage below. Once riscv64 supports DYAMIC_ARCH, this special case
can go away.
